flow/connectors/postgres/postgres.go

```diff
@@ -140,13 +141,29 @@ func ParseConfig(connectionString string, pgConfig *protos.PostgresConfig) (*pgx
 	return connConfig, nil
 }
 
-func (c *PostgresConnector) fetchCustomTypeMapping(ctx context.Context) (map[uint32]shared.CustomDataType, error) {
+func (c *PostgresConnector) fetchCustomTypeMapping(ctx context.Context, version uint32) (map[uint32]shared.CustomDataType, error) {
 	if c.customTypeMapping == nil {
 		customTypeMapping, err := shared.GetCustomDataTypes(ctx, c.conn)
 		if err != nil {
 			return nil, err
 		}
 		c.customTypeMapping = customTypeMapping
+
+		if version >= shared.InternalVersion_CompositeTypeAsTuple {
+			var compositeTypeNames []string
+			for _, typeData := range customTypeMapping {
+				if typeData.Type == 'c' && typeData.Delim == 0 { // Only composite types
+					compositeTypeNames = append(compositeTypeNames, typeData.Name)
+				}
+			}
+			types, err := c.conn.LoadTypes(ctx, compositeTypeNames)
```

Comment:
this is failing in CI for #3384
```
failed to fetch custom type mapping: failed to load composite types: Unknown field for composite type "geometry_dump":  field "geom" (OID 18053) is not already registered.
```
seems like there's a mess with having to make sure dependencies are registered before loading. Looking at `LoadTypes` this can probably be cleaned up to query catalog ourselves for composite types specifically & figure out how to register types in correct order